pkg/pillar/nireconciler/linux_reconciler.go

```diff
@@ -1270,12 +1277,51 @@ func (r *LinuxNIReconciler) getNIRouteInfo(niID uuid.UUID) (routes []types.IPRou
 				}
 			}
 		}
+		var ipVer types.AddressType
+		switch route.Type() {
+		case generic.IPv4RouteTypename:
+			ipVer = types.AddressTypeIPV4
+		case generic.IPv6RouteTypename:
+			ipVer = types.AddressTypeIPV6
+		}
+		// Avoid returning nil destination network.
+		// nil is used for route destination by netlink for default routes.
+		dstNet := route.Dst
+		if dstNet == nil {
+			switch route.Type() {
+			case generic.IPv4RouteTypename:
+				dstNet = ipv4Any
+			case generic.IPv6RouteTypename:
+				dstNet = ipv6Any
+			}
+		}
 		routes = append(routes, types.IPRouteInfo{
-			DstNetwork: route.Dst,
+			IPVersion:  ipVer,
+			DstNetwork: dstNet,
 			Gateway:    route.Gw,
 			OutputPort: portLL,
 			GatewayApp: appGW,
 		})
 	}
+	// Return routes in a deterministic and easy-to-read order.
+	// First IPv4 routes will be listed, then IPv6 routes.
+	// Inside the set of routes of the same IP version, default routes appear
+	// first for clarity, followed by other routes ordered by prefix length,
+	// with more specific routes (longer prefixes) listed before broader ones.
+	// This is at least how Linux lists the routes of a routing table.
+	sort.Slice(routes, func(i, j int) bool {
```

Comment:
The entire function is a good candidate for a Unit test, if you have time for that)

Reply:
Working on it

Reply:
Will you add it to this PR? Or we should not wait for it and it comes later?

Reply:
Done, added

Reply:
cool, thanks!